File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/DeleteCommentPayload.java
package jp.co.pluswing.mysqlsample.sample.payload;

public class DeleteCommentPayload {
    private DeleteCommentPayload(Boolean success) {
        this.success = success;
    }

    public final Boolean success;

    public static DeleteCommentPayload fail() {
        return new DeleteCommentPayload(false);
    }

    public static DeleteCommentPayload success() {
        return new DeleteCommentPayload(true);
    }
}

File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/DeleteCommentPayload.kt
package jp.co.pluswing.mysqlsample.sample.payload

data class DeleteCommentPayload private constructor(val success: Boolean) {
    companion object {
        fun fail(): DeleteCommentPayload {
            return DeleteCommentPayload(false)
        }

        fun success(): DeleteCommentPayload {
            return DeleteCommentPayload(true)
        }
    }
}